header-rewrite-filter/header_processor.h

```diff
@@ -14,29 +14,62 @@ namespace HeaderRewriteFilter {
 
 class Processor {
 public:
+  Processor() {}
   virtual ~Processor() {}
   virtual absl::Status parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start) { return absl::OkStatus(); }
-  virtual void executeOperation(Http::RequestOrResponseHeaderMap& headers) { }
 };
 
+class SetBoolProcessor : public Processor {
+public:
+  SetBoolProcessor() {}
+  virtual ~SetBoolProcessor() {}
+  virtual absl::Status parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start);
+  virtual absl::Status executeOperation();
+  const std::string& getBoolName() const { return bool_name_; }
+  const bool getResult(bool negate) const { return negate ? !result_ : result_; }
+
+private:
+  void setBoolName(absl::string_view bool_name) { bool_name_ = std::string(bool_name); }
+
+  const std::pair<std::string, std::string>& getStringsToCompare() const { return strings_to_compare_; }
+  void setStringsToCompare(std::pair<absl::string_view, absl::string_view> strings_to_compare);
+
+  Utility::MatchType getMatchType() const { return match_type_; }
+  void setMatchType(Utility::MatchType match_type) { match_type_ = match_type; }
+
+  std::string bool_name_;
+  std::pair<std::string, std::string> strings_to_compare_;
+  Utility::MatchType match_type_;
+  bool result_;
+};
+
+
+using SetBoolProcessorSharedPtr = std::shared_ptr<SetBoolProcessor>;
+using SetBoolProcessorMapSharedPtr = std::shared_ptr<std::unordered_map<std::string, SetBoolProcessorSharedPtr>>;
+
 class ConditionProcessor : public Processor {
 public:
+  ConditionProcessor() {}
   virtual ~ConditionProcessor() {}
   virtual absl::Status parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start);
-  bool executeOperation();
+  virtual absl::Status executeOperation(SetBoolProcessorMapSharedPtr bool_processors);
+
+  bool getConditionResult() const { return condition_; }
 
 private:
   std::vector<Utility::BooleanOperatorType> operators_;
   std::vector<std::pair<absl::string_view, bool>> operands_; // operand and whether that operand is negated
-  bool isTrue_; // Note: will be used when connecting ConditionProcessor to SetBoolProcessor
+  bool condition_;
```

Comment:
I think we'll run into some trouble if we start keeping track of state in the ConditionProcessor which is currently not scoped per request (e.g. multiple reqs being evaluated at the same time). Can you instead just pass the result to the caller? This can be done in a followup PR